tools/oprt2/pkg/ctxcopy/contextualcopy.go

```diff
@@ -0,0 +1,60 @@
+/*
+ *  Copyright 2025 Gravitational, Inc
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package ctxcopy
+
+import (
+	"context"
+	"io"
+)
+
+// Copy is a contextual version of [io.Copy]. If the context is cancelled, calls to
+// read and write functions will error, stopping the copy.
+// This retains all properties of [io.Copy], including support for [io.WriterTo] and
+// [io.ReaderFrom].
+func Copy(ctx context.Context, dst io.Writer, src io.Reader) (int64, error) {
```

Comment:
This feels like an antipattern to me.

> // If the context is cancelled, calls to
> // read and write functions will error, stopping the copy.

How are you satisfying this? It looks like me like the context is only checked before the read or write operation begins, but as soon as it starts we stop respecting context cancelation and enter a blocking call.